### PR TITLE
[CM-402] Date picker support in form template

### PR DESCRIPTION
**Pull request by Sunilkumarr**
## Summary
This PR is a date picker implementation only.

Starting Form UI :

IMG_0259

iOS 14:

Date:

IMG_0254

Date and time.

IMG_0261

iOS 13 and below:

Simulator Screen Shot - iPhone 11 - 2020-10-05 at 18 06 37

Simulator Screen Shot - iPhone 11 - 2020-10-05 at 18 06 19

Simulator Screen Shot - iPhone 11 - 2020-10-05 at 18 06 29

Selected values will be updated in form UI  after clicking on the `DONE` button 

Simulator Screen Shot - iPhone 11 - 2020-10-05 at 18 25 01

## Testing
Tested with selecting a date , date-time, and time checked in the UI and also in the submit button tap.

### CODE REVIEW

File: Sources/Views/ALKFormCell.swift [hunk 1]
```diff
@@ -55,6 +56,46 @@ class ALKFormCell: ALKChatBaseCell<ALKMessageViewModel>, UITextFieldDelegate {
...
+            return true
+        }
+
+        guard let datePickerVC = formDatePickerViewController,
+              let topVC = ALPushAssist().topViewController else {
+            return true
+        }
+
+        datePickerVC.modalPresentationStyle = .overCurrentContext
+        topVC.present(datePickerVC, animated: true, completion: nil)
```
Comment (mukeshthawani) [inline Sources/Views/ALKFormCell.swift:95]: We can let the parent VC handle the launch part. You can add a callback just like we've for the button tap.

File: Sources/Views/ALKFormCell.swift [hunk 2]
```diff
@@ -189,6 +233,64 @@ extension ALKFormCell: UITableViewDataSource, UITableViewDelegate {
...
+            }
+            let cell: ALKFormDateItemCell = tableView.dequeueReusableCell(forIndexPath: indexPath)
+            cell.item = dateSelectItem
+            cell.valueTextField.delegate = self
+            cell.valueTextField.tag = indexPath.section
+            if let timeInMillSecs = formData?.dateFields[indexPath.section] {
+                let formatter = DateFormatter()
+                formatter.dateFormat = "dd-MM-yyyy"
+                let date = Date(timeIntervalSince1970: TimeInterval(timeInMillSecs / 1000))
+                let stringDate = formatter.string(from: date)
```
Comment (mukeshthawani) [inline Sources/Views/ALKFormCell.swift:249]: Date formatting can be moved to the Date extension file, this way we can reuse it.

Same can be done for time.

File: Sources/Views/ALKFormDateItemCell.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+class ALKFormTimeItemCell: ALKFormDateBaseCell {
+    var item: FormViewModelItem? {
+        didSet {
+            guard let item = item as? FormViewModelTimeItem else {
+                return
+            }
+            nameLabel.text = item.label
+            valueTextField.attributedPlaceholder =
+                NSAttributedString(string: "hh:mm a")
```
Comment (mukeshthawani) [inline Sources/Views/ALKFormDateItemCell.swift:33]: Make sure this format works in different language/regions.

File: Sources/Views/ALKFormCell.swift [hunk 4]
```diff
@@ -55,6 +56,46 @@ class ALKFormCell: ALKChatBaseCell<ALKMessageViewModel>, UITextFieldDelegate {
...
+                                                                           datePickerMode: .time,
+                                                                           localizedStringFileName: localizedStringFileName)
+        case .date:
+            formDatePickerViewController = ALKFormDatePickerViewController(delegate: self,
+                                                                           messageKey: key,
+                                                                           position: textField.tag,
+                                                                           datePickerMode: .date,
+                                                                           localizedStringFileName: localizedStringFileName)
+        case .dateTimeLocal:
+            formDatePickerViewController = ALKFormDatePickerViewController(delegate: self,
```
Comment (mukeshthawani) [inline Sources/Views/ALKFormCell.swift:80]: We can move it down and just select the mode in the Switch statement.

File: Sources/Views/ALKFormCell.swift [hunk 5]
```diff
@@ -55,6 +56,46 @@ class ALKFormCell: ALKChatBaseCell<ALKMessageViewModel>, UITextFieldDelegate {
...
+                                                                           messageKey: key,
+                                                                           position: textField.tag,
+                                                                           datePickerMode: .dateAndTime,
+                                                                           localizedStringFileName: localizedStringFileName)
+        default:
+            return true
+        }
+
+        guard let datePickerVC = formDatePickerViewController,
+              let topVC = ALPushAssist().topViewController else {
```
Comment (mukeshthawani) [inline Sources/Views/ALKFormCell.swift:90]: I just saw we've a `topViewController()` helper in `ApplozicSwift`. We can use that and remove `import Applozic`.

Comment (Sunilkumarr) [reply Sources/Views/ALKFormCell.swift]: Removed import

File: Sources/Views/ALKFormCell.swift (hunk 4 above)
Comment (Sunilkumarr) [reply Sources/Views/ALKFormCell.swift]: Done

File: Sources/Views/ALKFormDateItemCell.swift (hunk 3 above)
Comment (Sunilkumarr) [reply Sources/Views/ALKFormDateItemCell.swift]: added the 24hr formate

File: Sources/Views/ALKFormCell.swift (hunk 2 above)
Comment (Sunilkumarr) [reply Sources/Views/ALKFormCell.swift]: Moved

File: Sources/Views/ALKFormCell.swift (hunk 1 above)
Comment (Sunilkumarr) [reply Sources/Views/ALKFormCell.swift]: Done

**Review by mukeshthawani — APPROVED** [review]

**Pull request merged**